Add explicit disallow feature

SimonC-Audigent opened the pull request:
There are some scenarios, where we want to check if the robots.txt is explicitly disallowing the UA (wildcard not included). This a common behaviour for ads crawlers such as  Google AdsBots: https://developers.google.com/search/docs/crawling-indexing/robots/create-robots-txt, which will disregards wildcards.
To support that scenario, I added an `explicit` parameter, to the `isDisallowed` method.

samclarke commented:
Thanks for the PR! That would be a useful feature to add.

Instead of adding it as a boolean, it might be better to make it a separate method instead. For example, the difference between:
```js
if (robots.isDisallowed(url, 'my-ua', true)) {
```
and:
```js
if (robots.isDisallowed(url, 'my-ua', false)) {
```
would require checking the documentation but something like:
```js
if (robots.isExplicitlyDisallowed(url, 'my-ua')) {
```
is a little more obvious.  

Also adding a parameter is not guaranteed to be backwards compatible. It's unlikely to break anything but there's always a possibility someone has done something like:
```js
let results = arrayOfUserAgents.map(robots.isAllowed.bind(robots, 'some ur'))
```
which would be broken if added due to map passing the index as a parameter.

samclarke commented on Robots.js:
```diff
@@ -422,16 +427,51 @@ Robots.prototype.getMatchingLineNumber = function (url, ua) {
+	return !(!rule || rule.allow);
 };
 
+Robots.prototype.isExplicitlyDisallowed = function(url, ua) {	
```
This seems to be re-implementing `_getRule()`, is there a reason it doesn't call `_getRule()` with `explicit` set to `true`?

Thinking about it, it might be worth adding an `isExplicitlyAllowed()` method that duplicates `isAllowed()` but uses the explicit rule, something like:
```js
Robots.prototype.isExplicitlyAllowed = function (url, ua) {
        // Not sure if we want to allow the '*' user agent here or not?
	if (ua === undefined) {
		throw new Error("User Agent must be specified in explicit mode")
	}

	var rule = this._getRule(url, ua, true);
	[...]
```
then `isExplicitlyDisallowed()` could negate that like `isDisallowed()` currently does with `isAllowed()`.

That would make the API the same between both versions and avoids any incompatibility risks. What do you think?

samclarke commented on Robots.js:
```diff
@@ -361,7 +361,7 @@ Robots.prototype.setPreferredHost = function (url) {
 };
 
-Robots.prototype._getRule = function (url, ua) {
+Robots.prototype._getRule = function (url, ua, explicit) {
```
This looks good :+1:

samclarke reviewed (changes requested):
Just had a chance to look into this a bit more and added a couple of comments. It definitely seems like a feature worth adding.

SimonC-Audigent commented:
I refactored it to use isExplicitlyDisallowed and keep isDisallowed with the same signature, (which is what i implemented at the beginning and change my mind... but forgot to remove it from the PR).
